File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/DQMCommand.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package b2daq.dqm.core;

import b2daq.nsm.NSMCommand;

/**
 *
 * @author tkonno
 */
public class DQMCommand extends NSMCommand {

    static public final DQMCommand UNKNOWN = new DQMCommand(0, "UNKNOWN");
    static public final DQMCommand DQMDIRGET = new DQMCommand(701, "DQMDIRGET");
    static public final DQMCommand DQMDIRSET = new DQMCommand(702, "DQMDIRSET");
    static public final DQMCommand DQMHISTGET = new DQMCommand(703, "DQMHISTGET");
    static public final DQMCommand DQMHISTSET = new DQMCommand(704, "DQMHISTSET");
    static public final DQMCommand DQMLISTGET = new DQMCommand(705, "DQMLISTGET ");
    static public final DQMCommand DQMLISTSET = new DQMCommand(706, "DQMLISTSET ");
    static public final DQMCommand DQMFILEGET = new DQMCommand(707, "DQMFILEGET");
    static public final DQMCommand DQMFILESET = new DQMCommand(708, "DQMFILESET");
    static public final DQMCommand DQMRESET = new DQMCommand(709, "DQMRESET");

    static private final DQMCommand[] CMD_LIST = new DQMCommand[]{
        DQMDIRGET, DQMDIRSET, DQMHISTGET, DQMHISTSET,
        DQMLISTGET, DQMLISTGET, DQMFILEGET, DQMFILESET, DQMRESET
    };

    static DQMCommand get(int id) {
        for (DQMCommand cmd : CMD_LIST) {
            if (id == cmd.getId()) {
                return cmd;
            }
        }
        return UNKNOWN;
    }

    static DQMCommand get(String label) {
        for (DQMCommand cmd : CMD_LIST) {
            if (label.matches(cmd.getLabel())) {
                return cmd;
            }
        }
        return UNKNOWN;
    }

    public DQMCommand() {
        super(UNKNOWN);
    }

    private DQMCommand(int id, String label) {
        super(id, label);
    }
}

File: daq/slc/dqm/src/DQMCommand.cc
#include "daq/slc/dqm/DQMCommand.h"

using namespace Belle2;

const DQMCommand DQMCommand::DQMDIRGET(701, "DQMDIRGET");
const DQMCommand DQMCommand::DQMDIRSET(702, "DQMDIRSET");
const DQMCommand DQMCommand::DQMHISTGET(703, "DQMHISTGET");
const DQMCommand DQMCommand::DQMHISTSET(704, "DQMHISTSET");
const DQMCommand DQMCommand::DQMLISTGET(705, "DQMLISTGET ");
const DQMCommand DQMCommand::DQMLISTSET(706, "DQMLISTSET ");
const DQMCommand DQMCommand::DQMFILEGET(707, "DQMFILEGET");
const DQMCommand DQMCommand::DQMFILESET(708, "DQMFILESET");
const DQMCommand DQMCommand::DQMRESET(709, "DQMRESET");

const DQMCommand& DQMCommand::operator=(const std::string& label) throw()
{
  if (NSMCommand::operator=(label) != Enum::UNKNOWN) return *this;
  else if (label == DQMDIRGET.getLabel()) *this = DQMDIRGET;
  else if (label == DQMDIRSET.getLabel()) *this = DQMDIRSET;
  else if (label == DQMHISTGET.getLabel()) *this = DQMHISTGET;
  else if (label == DQMHISTSET.getLabel()) *this = DQMHISTSET;
  else if (label == DQMLISTGET.getLabel()) *this = DQMLISTGET;
  else if (label == DQMLISTSET.getLabel()) *this = DQMLISTSET;
  else if (label == DQMFILEGET.getLabel()) *this = DQMFILEGET;
  else if (label == DQMFILESET.getLabel()) *this = DQMFILESET;
  else if (label == DQMRESET.getLabel()) *this = DQMRESET;
  else *this = Enum::UNKNOWN;
  return *this;
}

const DQMCommand& DQMCommand::operator=(int id) throw()
{
  if (NSMCommand::operator=(id) != Enum::UNKNOWN) return *this;
  else if (id == DQMDIRGET.getId()) *this = DQMDIRGET;
  else if (id == DQMDIRSET.getId()) *this = DQMDIRSET;
  else if (id == DQMHISTGET.getId()) *this = DQMHISTGET;
  else if (id == DQMHISTSET.getId()) *this = DQMHISTSET;
  else if (id == DQMLISTGET.getId()) *this = DQMLISTGET;
  else if (id == DQMLISTSET.getId()) *this = DQMLISTSET;
  else if (id == DQMFILEGET.getId()) *this = DQMFILEGET;
  else if (id == DQMFILESET.getId()) *this = DQMFILESET;
  else if (id == DQMRESET.getId()) *this = DQMRESET;
  else *this = Enum::UNKNOWN;
  return *this;
}

const DQMCommand& DQMCommand::operator=(const char* label) throw()
{
  if (label != NULL)  *this = std::string(label);
  else *this = Enum::UNKNOWN;
  return *this;
}

